src/stores/game.js

```diff
@@ -4,6 +4,15 @@ import { cloneDeep } from 'lodash';
 import { io } from '@/plugins/sails.js';
 import { sleep } from '../util/sleep';
 
+/**
+ * Throws an error if VITE_USE_GAMESTATE_API is enabled
+ */
+const checkGameStateApi = () => {
+  if (import.meta.env.VITE_USE_GAMESTATE_API === 'true') {
+    throw 'This action is not supported yet in GameState API';
+  }
+};
```

Comment:
While this is essentially what the issue states, with the refactoring introduced in #971, we have the opportunity to centralize the weirdness of maintaining the two apis into just the `makeSocketRequest()` function.

The unsolved nuance is how we manage api calls that have different forms to their urls in the GameState API.

All the move-making requests i.e. `draw`, `points` etc, are changing to use `POST /api/game/:gameId/move/<the_move_name>`

whereas the other endpoints related to games that are not move-making e.g. creating a game, joining, readying up, are not currently changing their urls. 

The end result of this PR should be that we can boot, build and run the tests in `VITE_USE_GAMESTATE_API` mode in that mode we can log in, create a game, join it, ready up and start the game, at which point the cards should deal using the old method of creating `card` records, but then we can't make any of the moves e.g. drawing etc because they error. The question now is how to manage making the non-move requests work as normal while making the move-making ones error in the correct fashion.

I think the mechanism that we want to use for causing the move-making requests to error is basically what's already in `makeSocketRequest()`: to conditionally request an updated URL and then the backend will simply 404 if that url isn't supported yet. Then the question is how to leave the other requests alone. The three ways I can think to do that are:

1. Add an argument to `makeSocketRequest()` that enables/disables the url transform and setting it appropriately in each caller
2. Updating `makeSocketRequest()` itself to check the requested `slug` against a known set of ones to mutate (all the move making slugs) and have that function itself determine whether to change the request based on whether the slug is found in its set of known ones to change
3. Splitting off a separate function for the transforming version of `makeSocketRequest()` e.g. `makeGameStateAPIRequest()`, putting the conditional logic in there, and calling `makeGameStateAPIRequest()` in the move-making functions and `makeSocketRequest()` in the others.


@Haviles04 @Plubedam penny for your thoughts. Once this is settled I can go through and update the issues to remove the section on fixing these frontend errors because it should come for 'free' with the backend implementing the appropriate endpoint.

Reply:
After giving it some thought, I decided to go with number 2.
```suggestion
const nonGameStateAPIMoves = new Set(['concede', 'stalemate', 'reject-stalemate', 'rematch']);
```
Instead of creating a Set of the move-making slugs, I used the non move-making ones so the list is much shorter.
```suggestion
      const usingGameStateAPI = import.meta.env.VITE_USE_GAMESTATE_API === 'true';
      let url = `/api/game/${slug}`;

      if (usingGameStateAPI && !nonGameStateAPIMoves.has(slug)) {
        url = `/api/gamestate/${this.id}/move/${slug}`;
        method = 'post';
      }
```
Then we check if the slug is not part of the Set and the env variable is true. Since all move-making slugs are POST we always set the `method`. After we implement all new GameState moves and it's time to normalize the rest of the routes, we probably should make the `method` parameter default to 'post', and define the minority of non POST endpoints `method` individually

Reply:
Doesn’t make sense to have a function per url, as getters essentially, that either return the new url or the old one? It’s a little more overhead to start but would allow you to opt in per route.

Reply:
@seriouslysean interesting point. An issue with one function per url is that each caller needs to call their correct function. I think what might instead be ideal is to create a url-transforming function that inputs a slug and outputs the correct url with a big switch statement for managing the different cases. 

At the moment, that should work equivalently to what is written here, but then we could add urls to the switch case to handle different urls differently when we get there. Something like this.

```
function transformGameUrl() {
  if (import.meta.env.VITE_USE_GAMESTATE_API !== 'true') {
    return  `/api/game/${slug}`;
  }

  switch (slug) {
    case 'pass':
    case 'draw':
    // add all the move-making ones here
      return `api/game/${this.gameId}/move/${slug};
    default: 
      return  `/api/game/${slug}`;
  }
}

```